src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs

```diff
@@ -15,98 +18,98 @@ namespace System.Xml.Serialization
     ///<internalonly/>
     public abstract class XmlSerializationReader : XmlSerializationGeneratedCode
     {
-        private XmlReader _r;
-        private XmlDocument _d;
-        private Hashtable _callbacks;
-        private Hashtable _types;
-        private Hashtable _typesReverse;
+        private XmlReader _r = null!;
```

Comment:
Why is this non-nullable? Based on the following snippet:
```cs
class MyXmlReader : XmlSerializationReader
{
    protected override void InitCallbacks()
    {
        throw new NotImplementedException();
    }

    protected override void InitIDs()
    {
        throw new NotImplementedException();
    }
}
```
You can access the `Reader` property on a `MyXmlReader` instance and see that is `null`.

Reply:
The reason this is non-null is because code is not expecting it to be null. You will get NRE on many methods. for derived classes you're supposed to call the Init method which initializes it with non-null but as you said it is not meant to be derived by users.

Reply:
>You will get NRE on many methods. for derived classes you're supposed to call the Init method which initializes it with non-null

I think is fine for this to be nullable if it will be `null` until `Init()` is called, that can make users aware of needing to call the method to avoid `NullReferenceException`s.

Reply:
at this point this will only cause lots of `!` in the code for no much benefit. it's private so derived classes won't see the annotation

Reply:
It's private but it is exposed through the `Reader` property which is protected. 
OK, since it's easier and this is not meant to be used by real world users, you can keep it as non-nullable.